types/object.go

```diff
@@ -5,6 +5,26 @@ import (
 	"sync/atomic"
 )
 
+// ObjectMode describes what users can operate on this object.
+//
+// Different object mode is orthogonal, an object could have different object mode
```

Comment:
Different object mode**s are** orthogonal, **and** an object could have different object mode**s** at the same time.